src/plan_partialize.c

```diff
@@ -174,17 +183,18 @@ partialize_agg_paths(RelOptInfo *rel)
  * simple filter (e.g., HAVING device > 3). In such cases, the HAVING clause is
  * removed and replaced by a filter on the input.
  * Returns : true if partial aggs were found, false otherwise.
+ * Modifies : output_rel if partials aggs were found.
  */
 bool
-ts_plan_process_partialize_agg(PlannerInfo *root, RelOptInfo *input_rel, RelOptInfo *output_rel)
+ts_plan_process_partialize_agg(PlannerInfo *root, RelOptInfo *output_rel)
 {
 	Query *parse = root->parse;
 	Assert(IS_UPPER_REL(output_rel));
 
 	if (CMD_SELECT != parse->commandType || !parse->hasAggs)
 		return false;
 
-	if (!has_partialize_function(parse))
+	if (!has_partialize_function(parse, TS_FIX_AGGREF))
```

Comment:
Shouldn't we check `rel->relatarget->expr` here instead of parse tree? Otherwise it will partialize for any rel.

Reply:
It is expected to partialize for any rel. I believe it can be changes to do it on `reltarget`, but imho this refactoring is not necessary for this PR.

Reply:
Does this check avoid doing things for, e.g., declarative partitioning? It seems to me that we will partialize any rel, including child partitions during partitionwise aggregation for declarative partitions, so I am wondering if this bug will still be hit for other tables than hypertables?

Reply:
I believe it will be hit on declarative partitioning. I created a separate internal issue to address it later.

Reply:
FWIW, it might be easy to fix by checking the extras parameter in the upper_rel hook? It contains a `GroupPathExtraData` struct that contains info on whether this is called for a partitionwise aggregation. We can just exclude calling this function in that case.

Reply:
Removing function from the target list is not an easy fix if I remember correctly.

Reply:
Is it necessary to remove it from the target list? Isn't it just a noop? Can we just clear the partitionwise paths so that we don't produce those plans in that case since they don't make sense anyway?

It is fine to leave for another PR I guess.